### ISSUE TITLE
Websocket failure connection issues on the F-Droid version client

### ISSUE THREAD

**Issue by alexdwork2018** [CTemplar/android#449]
https://sentry.io/organizations/ctemplar/issues/2908526132/?project=5462074
https://sentry.io/organizations/ctemplar/issues/2931544404/?project=5462074
https://sentry.io/organizations/ctemplar/issues/2923050349/?project=5462074

**Comment by The-Hidden-Hand** [comment]
There are two or three more that can be related in the first page.

**Comment by alexdwork2018** [comment]
@The-Hidden-Hand  Yes, one of them exactly relates to the first issue, perhaps others. I have added them above to follow.

**Comment by alexdwork2018** [comment]
https://github.com/CTemplar/android/commit/cf45283b71a499f2fca8c8f818751f0442b861bd
After researching the issue, I found out that manual Sentry capture point records each of the user's notification web socket failures.
I found out that manual Sentry logging records each of the user's web socket failures.. I repeated the issue on an emulator with parameters close to the issue described. Each web socket connection was failed a minute after initialization. After this client reconnected and all repeated many times. To prevent this I configured okhttp to automatic ping which keeps the connection alive. After testing I made sure this fix allow keeping connections without drops. All project libraries have been updated to the latest versions to include all the latest bug fixes. Libraries updates require the latest SDK so I checked and updated it too.
